### PR TITLE
PEP 394: Distributions can choose what does python mean

### PR DESCRIPTION
**Pull request by hroncok**

### CODE REVIEW

File: pep-0394.txt [hunk 1]
```diff
@@ -23,86 +25,92 @@ Python interpreter (i.e. the version invoked by the ``python`` command).
...
+Python 3.x ([5]_). As Python 2 reaches end of life in 2020 (PEP 373, PEP 404),
+distributors are moving away from Python 2, and having ``python`` point to
+Python 2.x gives false assumptions to the users about Python 2.x being the
+default. This PEP acknowledges that it is the distributors call whether to
+stay backwards compatible at this cost or move forward by aliasing the
+``python`` command to Python 3.x or not providing it at all.
+
+There is no more *universal* shebang that works anywhere, but the
+recommendations in this PEP provide the least bad option for upstreams:
+use ``python3`` shebangs if possible or avoid them.
```
Comment (hroncok) [inline pep-0394.txt]: @encukou I'd appreciate more ideas for rationale.

File: pep-0394.txt [hunk 2]
```diff
@@ -129,27 +137,24 @@ making such a change.
   and other users. Updating the ``python`` command to invoke ``python3``
   by default indicates that a distribution is willing to break such scripts
   with errors that are potentially quite confusing for users that aren't
-  yet familiar with the backwards incompatible changes in Python 3. For
+  familiar with the backwards incompatible changes in Python 3. For
   example, while the change of ``print`` from a statement to a builtin
   function is relatively simple for automated converters to handle, the
-  SyntaxError from attempting to use the Python 2 notation in versions of
-  Python 3 prior to 3.4.2 is thoroughly confusing if you aren't already
```
Comment (hroncok) [inline pep-0394.txt:136]: I don't consider `< 3.4.2` relevant here. Distributions stuck at `< 3.4.2` won't do this.

**Comment by hroncok** [comment]
Note that the `Application to the CPython Reference Interpreter` section would need updating but doing so would not document the reality. I suppose having a `make install` option for `/usr/bin/python` (keeping it opt-out on py2 and opt-in on py3) would be ideal. What is the proper procedure? Document the desired state in the PEP and provide a reference PRs for the code at the same time?

File: pep-0394.txt [hunk 3]
```diff
@@ -23,86 +25,92 @@ Python interpreter (i.e. the version invoked by the ``python`` command).
...
-  ``python`` on their shebang line.
+* Scripts that are deliberately written to be source compatible with both
+  Python 3.x and 2.x should use ``python3`` on their shebang line
+  (see `Rationale`_ for details).
+* One exception to this is scripts that are deliberately written to be used
+  on an *old system* where ``python3`` (or even ``ptyhon2``) is not available
+  (such as the default Python installed on macOS or RHEL 6).
+  Such scripts may continue to use ``python`` on their shebang line.
+* Scripts targeting both *old systems* and systems without the default
+  ``python`` command need to make a compromise a document this situation.
```
Comment (encukou) [inline pep-0394.txt]:
```suggestion
  ``python`` command need to make a compromise and document this situation.
```

**Comment by encukou** [comment]
I pushed suggestions directly here, do they look OK?

> Note that the `Application to the CPython Reference Interpreter` section would need updating but doing so would not document the reality. I suppose having a `make install` option for `/usr/bin/python` (keeping it opt-out on py2 and opt-in on py3) would be ideal. What is the proper procedure? Document the desired state in the PEP and provide a reference PRs for the code at the same time?

I'd leave that be. CPython's choices align with what a distributor is allowed to do.

**Comment by hroncok** [comment]
Awesome! Should we submit this as is?

**Comment by encukou** [comment]
Sure!

**Comment by hroncok** [comment]
https://github.com/python/peps/pull/989

**Pull request closed without merging**